C++ getting-started guide + better minimal example (#3989)
rerun-io/rerun: examples/cpp/minimal/CMakeLists.txt

    FetchContent_Declare(rerun_sdk DOWNLOAD_EXTRACT_TIMESTAMP YES URL ${RERUN_CPP_URL})
    FetchContent_MakeAvailable(rerun_sdk)

>     # Rerun requires at least C++17, but you can use an even newer version.

Reviewer: ```suggestion
    # Rerun requires at least C++17, but it should be compatible with newer versions.
```
